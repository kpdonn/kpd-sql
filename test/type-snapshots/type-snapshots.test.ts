import { typingsTestDir, check } from "./typeChecker"
import * as path from "path"
import * as fs from "fs"

describe("Verify output of typings test files matches expected", async () => {
  const files = fs
    .readdirSync(typingsTestDir)
    .filter(it => it.endsWith(".ts"))
    .map(file => path.resolve(typingsTestDir, file))

  files.forEach(file =>
    it(`Verify output of typings test file ${path.basename(file)}`, async () => {
      const diagnostics = check(file)
      expect(diagnostics).toMatchSnapshot()
    })
  )
})
