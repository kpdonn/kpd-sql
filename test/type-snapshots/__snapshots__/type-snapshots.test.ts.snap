// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Verify output of typings test files matches expected Verify output of typings test file expect-errors.ts 1`] = `
Array [
  Object {
    "category": 1,
    "character": 12,
    "code": 2345,
    "file": "expect-errors.ts",
    "line": 10,
    "message": "Argument of type '[TableColumn<\\"class\\", NumberType, \\"semesterId\\", false, number>]' is not assignable to parameter of type 'ReadonlyArray<BaseColumn<string, Type<any, any, any>, string, boolean, any>> & { \\"0\\": never; } & { \\"0\\": any; }'.
  Type '[TableColumn<\\"class\\", NumberType, \\"semesterId\\", false, number>]' is not assignable to type '{ \\"0\\": never; }'.
    Types of property '\\"0\\"' are incompatible.
      Type 'TableColumn<\\"class\\", NumberType, \\"semesterId\\", false, number>' is not assignable to type 'never'.",
  },
  Object {
    "category": 1,
    "character": 16,
    "code": 2345,
    "file": "expect-errors.ts",
    "line": 23,
    "message": "Argument of type '[TableColumn<\\"myWith\\", NumberType, \\"id\\", false, number>, TableColumn<\\"myWith\\", NumberType, \\"courseId\\", false, number>]' is not assignable to parameter of type 'ReadonlyArray<BaseColumn<string, Type<any, any, any>, string, boolean, any>> & { \\"0\\": BaseColumn<\\"myWith\\", Type<any, any, any>, string, boolean, any>; \\"1\\": never; } & { \\"0\\": any; }'.
  Type '[TableColumn<\\"myWith\\", NumberType, \\"id\\", false, number>, TableColumn<\\"myWith\\", NumberType, \\"courseId\\", false, number>]' is not assignable to type '{ \\"0\\": BaseColumn<\\"myWith\\", Type<any, any, any>, string, boolean, any>; \\"1\\": never; }'.
    Types of property '\\"1\\"' are incompatible.
      Type 'TableColumn<\\"myWith\\", NumberType, \\"courseId\\", false, number>' is not assignable to type 'never'.",
  },
  Object {
    "category": 1,
    "character": 12,
    "code": 2345,
    "file": "expect-errors.ts",
    "line": 29,
    "message": "Argument of type '[TableColumn<\\"class\\", NumberType, \\"semesterId\\", false, number>, TableColumn<\\"semester\\", NumberType, \\"year\\", false, number>]' is not assignable to parameter of type 'ReadonlyArray<BaseColumn<string, Type<any, any, any>, string, boolean, any>> & { \\"0\\": BaseColumn<\\"class\\", Type<any, any, any>, string, boolean, any>; \\"1\\": BaseColumn<\\"class\\", Type<any, any, any>, string, boolean, any>; } & { \\"0\\": any; }'.
  Type '[TableColumn<\\"class\\", NumberType, \\"semesterId\\", false, number>, TableColumn<\\"semester\\", NumberType, \\"year\\", false, number>]' is not assignable to type '{ \\"0\\": BaseColumn<\\"class\\", Type<any, any, any>, string, boolean, any>; \\"1\\": BaseColumn<\\"class\\", Type<any, any, any>, string, boolean, any>; }'.
    Types of property '\\"1\\"' are incompatible.
      Type 'TableColumn<\\"semester\\", NumberType, \\"year\\", false, number>' is not assignable to type 'BaseColumn<\\"class\\", Type<any, any, any>, string, boolean, any>'.
        Types of property '_tableAs' are incompatible.
          Type '\\"semester\\"' is not assignable to type '\\"class\\"'.",
  },
  Object {
    "category": 1,
    "character": 12,
    "code": 2345,
    "file": "expect-errors.ts",
    "line": 34,
    "message": "Argument of type '[TableColumn<\\"class\\", NumberType, \\"semesterId\\", false, number>, TableColumn<\\"class\\", NumberType, \\"semesterId\\", false, number>]' is not assignable to parameter of type 'ReadonlyArray<BaseColumn<string, Type<any, any, any>, string, boolean, any>> & { \\"0\\": NoDuplicates<\\"semesterId\\">; \\"1\\": NoDuplicates<\\"semesterId\\">; } & { \\"0\\": any; }'.
  Type '[TableColumn<\\"class\\", NumberType, \\"semesterId\\", false, number>, TableColumn<\\"class\\", NumberType, \\"semesterId\\", false, number>]' is not assignable to type '{ \\"0\\": NoDuplicates<\\"semesterId\\">; \\"1\\": NoDuplicates<\\"semesterId\\">; }'.
    Types of property '\\"0\\"' are incompatible.
      Type 'TableColumn<\\"class\\", NumberType, \\"semesterId\\", false, number>' is not assignable to type 'NoDuplicates<\\"semesterId\\">'.
        Property '\\"NO DUPLICATE KEYS ALLOWED\\"' is missing in type 'TableColumn<\\"class\\", NumberType, \\"semesterId\\", false, number>'.",
  },
  Object {
    "category": 1,
    "character": 29,
    "code": 2345,
    "file": "expect-errors.ts",
    "line": 40,
    "message": "Argument of type '\\"wrong type\\"' is not assignable to parameter of type 'number | PlaceholderParam<string> | TableColumn<string, Type<any, any, any>, string, boolean, number> | Aggregate<string, Type<any, any, any>, string, number> | null'.",
  },
  Object {
    "category": 1,
    "character": 15,
    "code": 2554,
    "file": "expect-errors.ts",
    "line": 42,
    "message": "Expected 1 arguments, but got 0.",
  },
  Object {
    "category": 1,
    "character": 12,
    "code": 2345,
    "file": "expect-errors.ts",
    "line": 54,
    "message": "Argument of type '{ myTest: string; }' is not assignable to parameter of type 'Record<\\"myTest\\", number>'.
  Types of property 'myTest' are incompatible.
    Type 'string' is not assignable to type 'number'.",
  },
  Object {
    "category": 1,
    "character": 35,
    "code": 2345,
    "file": "expect-errors.ts",
    "line": 62,
    "message": "Argument of type 'string' is not assignable to parameter of type 'never'.",
  },
  Object {
    "category": 1,
    "character": 28,
    "code": 2345,
    "file": "expect-errors.ts",
    "line": 69,
    "message": "Argument of type 'TableColumn<\\"course\\", NumberType, \\"subjectId\\", false, number>' is not assignable to parameter of type 'string | PlaceholderParam<string> | TableColumn<string, Type<any, any, any>, string, boolean, string> | Aggregate<string, Type<any, any, any>, string, string>'.
  Type 'TableColumn<\\"course\\", NumberType, \\"subjectId\\", false, number>' is not assignable to type 'Aggregate<string, Type<any, any, any>, string, string>'.
    Property '_aggColumn' is missing in type 'TableColumn<\\"course\\", NumberType, \\"subjectId\\", false, number>'.",
  },
  Object {
    "category": 1,
    "character": 14,
    "code": 2345,
    "file": "expect-errors.ts",
    "line": 79,
    "message": "Argument of type '{ wrongKey: number; }' is not assignable to parameter of type 'Record<\\"myTest\\", number>'.
  Object literal may only specify known properties, and 'wrongKey' does not exist in type 'Record<\\"myTest\\", number>'.",
  },
  Object {
    "category": 1,
    "character": 12,
    "code": 2345,
    "file": "expect-errors.ts",
    "line": 86,
    "message": "Argument of type '{ myTest: null; }' is not assignable to parameter of type 'Record<\\"myTest\\", number>'.
  Types of property 'myTest' are incompatible.
    Type 'null' is not assignable to type 'number'.",
  },
]
`;

exports[`Verify output of typings test files matches expected Verify output of typings test file no-errors.ts 1`] = `Array []`;

exports[`Verify output of typings test files matches expected Verify output of typings test file tables.ts 1`] = `Array []`;
